rootfs/etc/confd/templates/builder

```diff
@@ -136,11 +139,19 @@ kubectl --namespace=$POD_NAMESPACE create -f /etc/${SLUG_NAME}.yaml
 
 #check for image creation or slug existence in S3EP
 
-while [ ! -f /apps/${SLUG_NAME}/slug.tgz ]
-do
-  sleep 2
-  echo "no file"
-done
+if [[ "$REMOTE_STORAGE" == "1" ]]; then
+  LS_CMD="$MC_PREFIX ls $PUSH_URL"
+  until $LS_CMD &> /dev/null; do
+    echo "no file"
+    sleep 2
+  done
+else
+  while [ ! -f /apps/${SLUG_NAME}/slug.tgz ]
+  do
+    sleep 2
+    echo "no file"
+  done
```

Comment:
@technosophos fyi I changed this in https://github.com/deis/builder/commit/36607fa4d1c518374ae7c6b1bb0fcf70025f6a02. It now continuously appends `.` characters to the end of the line, which is no less useful than what we have now (and takes up less screen real estate). My hope is that this will become better in the future.